src/Dhall.hs

```diff
@@ -505,20 +506,34 @@ defaultInterpretOptions = InterpretOptions
     for automatically deriving a generic implementation
 -}
 class GenericInterpret f where
-    genericAutoWith :: InterpretOptions -> Type (f a)
+    genericAutoWith :: InterpretOptions -> State Int (Type (f a))
```

Comment:
The amended commit much simplified the diff. IMHO Reader may be unnecessary anymore...